pkg/rpaas/client/client.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2019 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type RpaasClient struct {
+	hostAPI      string
+	tsuruTarget  string
+	tsuruToken   string
+	target       string
+	tsuruService string
+	httpClient   *http.Client
+}
+
+func New(hostAPI string) *RpaasClient {
+	client := &RpaasClient{httpClient: &http.Client{}, hostAPI: hostAPI}
+	return client
+}
+
+func NewTsuruClient(tsuruAPI, service, token string) (*RpaasClient, error) {
+	if tsuruAPI == "" || token == "" || service == "" {
+		return nil, fmt.Errorf("cannot create client without either tsuru target, token or service")
+	}
+	return &RpaasClient{
+		httpClient:   &http.Client{},
+		tsuruTarget:  tsuruAPI,
+		tsuruToken:   token,
+		tsuruService: service,
+	}, nil
+}
+
+type flavor struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type plan struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Default     bool   `json:"default"`
+}
+
+func (c *RpaasClient) Plans(ctx context.Context, instance *string) ([]plan, error) {
+
+	var pathName string
+	switch instance {
+	case nil:
+		pathName = fmt.Sprintf("/resources/plans")
+	default:
+		pathName = fmt.Sprintf("/resources/%s/plans", *instance)
+	}
+
+	req, err := c.newRequest("GET", *instance, pathName, nil)
```

Comment:
What does happen here when instance is nil?

Reply:
nice catch